flask_simplelogin/__init__.py

```diff
@@ -180,9 +197,19 @@ def init_app(self, app, login_checker=None,
         if login_form:
             self._login_form = login_form
 
+        # If the user is passing a new dictionary
         if messages and isinstance(messages, dict):
             self.messages.update(messages)
-
+        # If the user is disabling messages
+        # Must differentiate between None and False.
+        elif messages is False:
+            
+            # I almost put this in a dict comprehension, but I couldn't figure out how to update value.enabled
+            disabled_messages={}
+            for key, value in self.messages.items():
+                value.enabled=False
+                disabled_messages[key]=value
+            self.messages.update(disabled_messages)
```

Comment:
I think this block could be simply an iteration over `Message` instances:

```python
for message in self.messages.values():
    message.enabled = False
```

Have you tried something like that? Let me know if I'm missing anything and I can help in the design here ; )

Reply:
Will this actually update `self.messages`? It looks to me like this is just iterating through, throwing `message` away after each iteration. Am I missing something? How do we update `self.messages` from here?

Reply:
Well... I think this is better shown than explained, but long story short, `self.messages` just keeps a reference to the `Message` instances, so you actually don't need to change `self.messages`, but the objects it will look at:

```python
class Message:
  def __init__(self, enabled):
      self.enabled = enabled

messages = {
  42: Message(enabled=True),
  "42": Message(enabled=False)
}

for message in messages.values():
  message.enabled = True

print(messages[42].enabled)
print(messages["42"].enabled)
```

Will output:

```
True
True
```